src/core/providers-strategy/WalletConnectProviderStrategy.ts

```diff
@@ -0,0 +1,276 @@
+      onClientEvent: handleOnEvent
+    };
+
+    try {
+      const {
+        walletConnectV2ProjectId,
+        walletConnectV2Options = {},
+        walletConnectV2RelayAddress = ''
+      } = config;
+      const walletConnectProvider = new WalletConnectV2Provider(
+        providerHandlers,
+        chainId,
+        walletConnectV2RelayAddress,
+        walletConnectV2ProjectId,
+        walletConnectV2Options
+      );
+
+      await walletConnectProvider.init();
+
+      return { walletConnectProvider, dappMethods };
+    } catch (err) {
+      console.error('Could not initialize walletConnect', err);
+
+      if (isLoggedIn) {
+        await provider.logout();
+      }
+
+      throw err;
+    }
+  };
+
+  private login = async (options?: {
+    callbackUrl?: string;
+    token?: string;
+  }): Promise<{
+    address: string;
+    signature: string;
+  }> => {
+    if (!this.provider || !this.manager) {
+      throw new Error(
+        'Provider or manager is not initialized. Call createProvider first.'
+      );
+    }
+
+    const isConnected = this.provider.isConnected();
+
+    if (isConnected) {
+      throw new Error(WalletConnectV2Error.connectError);
+    }
+
+    const reconnect = async (): Promise<{
+      address: string;
+      signature: string;
+    }> => {
+      if (!this.provider || !this.manager) {
+        throw new Error(
+          'Provider or manager is not initialized. Call createProvider first.'
+        );
+      }
+
+      try {
+        await this.provider.init();
+
+        const { uri = '', approval: wcApproval } = await this.provider.connect({
+          methods: this.methods
+        });
+
+        this.manager.updateWcURI(uri);
+
+        const providerInfo = await this.provider.login({
+          approval: wcApproval,
+          token: options?.token
+        });
+
+        const { address = '', signature = '' } = providerInfo ?? {};
+
+        this.manager.closeAndReset();
+        this.unsubscribeEvents?.();
+        return { address, signature };
+      } catch {
+        return await reconnect();
+      }
+    };
+
+    if (!this.approval) {
+      throw new Error(
+        'Approval is not initialized. Call createProvider first.'
+      );
+    }
+
+    try {
+      const providerData = await this.provider.login({
+        approval: this.approval,
+        token: options?.token
+      });
+
+      const { address = '', signature = '' } = providerData ?? {};
+
+      this.manager.closeAndReset();
+      this.unsubscribeEvents?.();
+      return { address, signature };
+    } catch (error) {
+      console.error(WalletConnectV2Error.userRejected, error);
+      return await reconnect();
+    }
+  };
+
+  private logout = async (): Promise<boolean> => {
+    if (!this.provider) {
+      throw new Error(
+        'Provider is not initialized. Call createProvider first.'
+      );
+    }
+
+    try {
+      await logout({ provider: this.provider as unknown as IProvider });
```

Comment:
what happened to DappProvider? it's there that logout should be called like this